src/reportengine/api.py

```diff
@@ -0,0 +1,45 @@
+"""
+API for resource engine
+"""
+
+import logging
+import importlib
+
+from reportengine.resourcebuilder import ResourceBuilder
+from reportengine.resourcebuilder import FuzzyTarget
+from reportengine.configparser import Config
+from reportengine.environment import Environment
+
+log = logging.getLogger(__name__)
+
+class API:
+    """The API class"""
+    config_class = Config
+    environment_class = Environment
+
+    def __init__(self, providers, **kwargs):
+        prov_list = []
+        for prov in providers:
+            try:
+                mod = importlib.import_module(prov)
```

Comment:
Could we try to import the thing only if it is a string. I could think on situations (like currently e.g. the tests of reportengine) where you might want to have a class acting as a provider module.